### DISCUSSION TITLE
Diff command implementation

### DISCUSSION THREAD

**Discussion by alexmaslenn** [xlab-si/xopera-opera#139] [Show and tell]
Implementation of diff command for Topology Templates
Currently works on Template, not Instance level
https://github.com/xlab-si/xopera-opera/tree/diff-update

Connected issues #138 #130

**Comment by dradX** [comment]
Excellent @alexmaslenn! As we discussed before we should also support the instance level diffs. But I find the option to calculate template diffs very useful.

**Comment by anzoman** [comment]
#147 has been merged and the [dev opera release](https://test.pypi.org/project/opera/0.6.4.dev3/) is already available on Test PyPI instance.

**Comment by anzoman** [comment]
I want to add an additional question - will we also extend the support for opera `diff` and `update` commands to work on (compressed) TOSCA CSARs or is that not a current priority?

**Reply by alexmaslenn** [reply]
Yes we will.

**Comment by anzoman** [comment]
#156 has been merged too. The newest `opera update` command is available within the latest opera pre-release on Test PyPI here: https://test.pypi.org/project/opera/0.6.4.dev8/.

**Comment by anzoman** [comment]
Both `opera diff` and `opera update` commands are now implemented and available in the latest opera version - [0.6.4](https://pypi.org/project/opera/0.6.4/).
